### ISSUE TITLE
Question regarding Example 5 - Exhaustive Feature Selection and GridSearch

### ISSUE THREAD

**Issue by Micha-93** [rasbt/mlxtend#663]
Hello,
i have two questions regarding Example 5 - Exhaustive Feature Selection and GridSearch.

1) Does one not overestimate the performance of the model if we use the same trainset (even if they are splitted by two different cv-generators) for feature selection and hyperparameter optimization?

2) I want to use this simultanious feature selection and hyperparameter optimization in regression. How does the comparison in neg-mse-score or r2 score take into account the better scores simply by the fact that more features are used to train a model? (Dont we need the adjusted r2 here?)

**Comment by rasbt** [comment]
Thanks for your feedback, these are good points to consider.

> 1	Does one not overestimate the performance of the model if we use the same trainset (even if they are splitted by two different cv-generators) for feature selection and hyperparameter optimization?

Yeah, but that's unavoidable except you have a fresh, indepedent validation set for each feature combination. There's always going to be some bias

> 2	I want to use this simultanious feature selection and hyperparameter optimization in regression. How does the comparison in neg-mse-score or r2 score take into account the better scores simply by the fact that more features are used to train a model? (Dont we need the adjusted r2 here?)

Good point, that's why I didn't show the R2 in the examples at first. Due to how the sklearn API works, the adjusted R2 cannot be computed as a "scorer" function. However, I just added an example to the docs that illustrate a workaround. 

See Example 4 in https://github.com/rasbt/mlxtend/blob/master/docs/sources/user_guide/feature_selection/ExhaustiveFeatureSelector.ipynb

Let me know if you have any feedback or questions.
